clarifai/rag.py

```diff
@@ -0,0 +1,122 @@
+from datetime import datetime
+
+import yaml
+from google.protobuf.struct_pb2 import Struct
+
+from clarifai.client.app import App
+from clarifai.client.model import Model
+from clarifai.client.user import User
+from clarifai.client.workflow import Workflow
+
+
+class RAG:
+  """
+    RAG is a class for Retrieval Augmented Generation.
+
+    Example:
+        >>> from clarifai.rag import RAG
+        >>> rag_agent = RAG()
+    """
+
+  def __init__(self,
+               workflow_url: str = None,
+               base_url: str = "https://api.clarifai.com",
+               pat: str = None,
+               **kwargs):
+    """
+        Initialize an empty or existing RAG.
+        """
+    w = Workflow(url=workflow_url, base_url=base_url, pat=pat)
+    self._prompt_workflow = w
+    self._app = App(app_id=w.app_id, base_url=w.base, pat=w.pat)
+    del w
+
+  def setup(self,
```

Comment:
We could have this as a `classmethod` instead
```python
rag = RAG.setup()
```

Reply:
> init would take care of setting up a new RAG setup

 `setup()` takes `base_workflow`, `llm_url` etc as args. Not sure if we could merge with init and make `from_workflow(url)` work

Reply:
I think I get what you mean now. Will try out your suggestion